op-service/event/system.go

```diff
@@ -80,6 +92,34 @@ type systemActor struct {
 	emitPriority Priority
 }
 
+func (r *systemActor) traceAndLogEventEmitted(ctx context.Context, ev Event) context.Context {
+	_, path, line, _ := runtime.Caller(2)
+	if strings.Contains(path, "limiter.go") {
+		_, path, line, _ = runtime.Caller(3)
+	}
+
+	file := filepath.Base(path)
+	dir := filepath.Base(filepath.Dir(path))
+
+	var euuid string
+	var estep int
+	if ctx.Value(ctxKeyEventUuid) == nil {
+		euuid = uuid.New().String()[:6]
+		estep = 0
+		ctx = context.WithValue(ctx, ctxKeyEventUuid, euuid)
+		ctx = context.WithValue(ctx, ctxKeyEventStep, estep)
+	} else {
+		euuid = ctx.Value(ctxKeyEventUuid).(string)
+		estep = ctx.Value(ctxKeyEventStep).(int)
+		estep++
+		ctx = context.WithValue(ctx, ctxKeyEventStep, estep)
+	}
+
+	r.sys.log.Trace("Event emitted", "euid", fmt.Sprintf("%s:%d", euuid, estep), "ev", ev.String(), "loc", fmt.Sprintf("%s/%s:%d", dir, file, line))
```

Comment:
For `fmt.Sprintf("%s:%d", euuid, estep)`, may implement a stringer method like
```go
type eventTrace struct {
    UUID string
    Step int
}

func (et *eventTrace) String() string {
    return fmt.Sprintf("trace[%s:%d]", et.UUID, et.Step)
}
```

Also `ev.String()`'s `String()` is not necessary.

Reply:
I really want to keep the line length minimal, so I'm against the `trace[]` part.

I will convert `ev.String()` to `ev`.

Let me know if you feel strongly that we should add a type around event trace.